### DISCUSSION TITLE
Support for SolidJS

### DISCUSSION THREAD

**Discussion by dmahajan980** [chakra-ui/chakra-ui#5463] [Ideas]
### Description

This feature request is intended to discuss for adding Chakra UI support for [SolidJS](https://www.solidjs.com/) and to know about the core team's opinions in this regard.

### Problem Statement/Justification

Chakra UI has worked amazingly well for React web apps. [SolidJS](https://www.solidjs.com/) is an amazing framework that shares the same concepts and abstractions as React but is based on reactivity. For example, SolidJS also uses JSX and component model to build UI.

Given the growth of SolidJS and the community's expanding interest in using Solid, I would like to propose adding support for SolidJS. I am available to help in bridging this amazing library for SolidJS.

### Proposed Solution or API

Ideally, the APIs and components would be the same as those in Chakra UI React. However, those could be different if Solid's API impose any limitations.

**Comment by thedanchez** [comment]
Yes! I absolutely 2nd this request and would like to see that support for Solid as well!

**Comment by fabien-ml** [comment]
Hi, i'm working on a SolidJS component library heavily inspired by Chakra UI with _almost_ the same api, currently it has: 
- Style props (with responsive props support)
- Over 40 components
- Dark mode
- Theming
- Follow ARIA 1.2 design patterns

It's a work in progress so any feedback are welcome, the documentation is available at [hope-ui.com](https://hope-ui.com).

**Reply by bilalmohib** [reply]
Great Work Sir

**Reply by DmacMcgreg** [reply]
For anyone reading this, it seems like the above author changed his mind, and is removing the chakra-ui-esque api/syntax in a later version.

**Reply by fabien-ml** [reply]
Yes with the release of [zag-js](https://zagjs.com/) and the intent of chakra ui team to have a SolidJS version, it make no sense to be a second zone copy of Chakra

**Reply by DmacMcgreg** [reply]
Do you have a reference to where the team mentioned they were building a solid-js version by chance?

**Reply by fabien-ml** [reply]
In the experiment section https://github.com/orgs/chakra-ui/projects/3/views/3, however it's just a word on a spreadsheet for now.

**Reply by futhr** [reply]
@fabien-ml Nice work, but you actually state that [Hope UI is moving away from JSS](https://github.com/hope-ui/hope-ui/discussions/168) so your UI lib will be out of context in v1.0 as a direct alternative if we wish to port to SolidJS.

_I'm working on a white label multi-tenancy product, and a JSS based UI like Chakra UI was a revolution for these type of products._

**Reply by fabien-ml** [reply]
@futhr sorry i've forgot to share update on this thread, after reflexion i've decided to keep the library "as is" with style props and the "Chakra UI like API" because it's how people like using it (more info on the up to date discussion here https://github.com/hope-ui/hope-ui/discussions/176). sorry for the communication mess 😅

**Comment by foreturn** [comment]
Hi，any process?
